webserver/webserver/src/main/java/io/helidon/webserver/WebServer.java

```diff
@@ -143,6 +143,27 @@ default int port() {
      */
     int port(String socketName);
 
+    /**
+     * Update TLS configuration of the default socket {@link WebServer#DEFAULT_SOCKET_NAME}.
+     *
+     * If the socket has not been https and now the TLS is being configured, an exception will be thrown.
```

Comment:
I think it might be better to document exceptions with`\@throws` event if they are runtime exceptions, and add them to the signature.

How about simplifying the errors?:
- require a non null returned from `tls.sslContext()`
- if the current sslContext is null throw an `IllegalStateException` with "TLS context is not set, update not allowed"

The exceptions from HttpInitializer could remain, but this would be part of the WebServer contract and would be implemented by `NettyWebServer`.

Reply:
I have used tls.enabled() instead of tls.sslContext() since sslContext() is set to null only if the TLS is not enabled. Otherwise it makes sense.